### PR TITLE
WIP: PixyCam

### PR DESCRIPTION
**Pull request by ashwinc12**
The PixyCam line follow seemed to be successful on Frankie, so the following need to be completed for it to work on Apollo

- [ ] Adjust camera settings to highly detect only the gaffers tape. This can be done directly on the Pixy using PixyMon. Also in the arduino side, there's a commented out line of code that sets the Pixy up to detect white lines.

- [ ] Figure out Pixy mounting situation. The Pixy needs to be in an angled bracket of some kind in the middle of the robot. A CAD for this should be made soon.

- [ ] Adjust constant values for Apollo. Currently, they are only 3 numbers that we will have to adjust for this to work on Apollo. 1.) The error threshold constant 2.) Normal driving velocity 3.) Correctional velocity

### CODE REVIEW

**Comment by ashwinc12** [comment]
We can remove them when we go to test the actual bot

**Comment by roshanr10** [comment]
for future reference, if you keep all of those changes in a separate commit, then you can just revert one commit as opposed to cherrypicking the changes. just an FYI

File: src/main/java/frc/team4909/robot/PixyFollow.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        System.out.println("Command is called");
+    }
+    @Override
+    protected void execute() {
+        
+        Robot.arduino.writeString(str); 
+        byte[] arr = Robot.arduino.read(1);
+
+        if(arr[0] == 1){ // case 1 does not find line
+            Robot.myDrive.tankDrive(0.2, 0.2);
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/PixyFollow.java]: constants pls

File: arduinofollow/PixyFollow.ino [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  Serial.print("Starting...\n");
+  pixy.init();
+  // Turn on both lamps, upper and lower for maximum exposure
+  pixy.setLamp(1, 1);
+  // change to the line_tracking program.  Note, changeProg can use partial strings, so for example,
+  // you can change to the line_tracking program by calling changeProg("line") instead of the whole
+  // string changeProg("line_tracking")
+  pixy.changeProg("line");
+//  pixy.setMode(LINE_MODE_WHITE_LINE); should implement later
+  a = 1;
```
Comment (roshanr10) [inline arduinofollow/PixyFollow.ino]: could these be given more descriptive names?

File: src/main/java/frc/team4909/robot/Robot.java [hunk 3]
```diff
@@ -99,117 +105,124 @@ public static double map(double value, double old_min, double old_max, double ne
...
-    c.setClosedLoopControl(true); // Start Compressor in Closed Loop Control
+  //   // Compressor
+  //   c = new Compressor(0); // Initialize Compressor
+  //   c.setClosedLoopControl(true); // Start Compressor in Closed Loop Control
+    m_left = new Spark(0);
+    m_left.setSafetyEnabled(false);
+    m_right = new Spark(1);
+    m_right.setSafetyEnabled(false);
+    myDrive = new DifferentialDrive(m_left, m_right);
+    arduino = new SerialPort(115200, SerialPort.Port.kUSB);
```
Comment (roshanr10) [inline src/main/java/frc/team4909/robot/Robot.java]: constants

**Comment by ashwinc12** [comment]
Doesn't seem like this is going to be used this year; a good starting point for future implementation

**Pull request closed without merging**